Lib/unittest/mock.py

```diff
@@ -2482,6 +2485,55 @@ def __set__(self, obj, val):
         self(val)
 
 
+class WaitableMock(MagicMock):
+    """
+    A mock that can be used to wait until it was called.
+
+    `event_class` - Class to be used to create event object.
+    Defaults to Threading.Event and can take values like multiprocessing.Event.
+    """
+
+    def __init__(self, *args, event_class=threading.Event, **kwargs):
+        _safe_super(WaitableMock, self).__init__(*args, **kwargs)
+        self._event = event_class()
+        self._expected_calls = defaultdict(lambda: event_class())
```

Comment:
You can probably simplify this to be just ‘defaultdict(event_class)’. Without the lambda.

Reply:
Removed. Thanks 👍